shared/js/background/message-handlers.js

```diff
@@ -197,6 +197,16 @@ const isExpectedSender = (sender) => (
     sender.url.match(/^https:\/\/(([a-z0-9-_]+?)\.)?duckduckgo\.com\/email/)
 )
```

Comment:
Can we make this more robust, I think this becomes more important if sites can log the user out. Also we've seen RegExp fail enough times we should probably avoid.

How about something like (untested):
```suggestion
function isExpectedEmailSender (sender) (
    try {
        let senderURL = new URL(sender.url)
        let domain = tldts.parse(sender.url).domain
        if (domain === 'duckduckgo.com' &&
            senderURL.pathname.startsWith('/email')) {
            return true
        }
    } catch {}
    
    return false
)
```

Reply:
@jonathanKingston Good shout -- updated 👍